modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/application/ApplicationExtensionsManagerConfigurationBuilder.java

```diff
@@ -53,17 +66,49 @@ protected void doConfigure(MuleContext muleContext) throws Exception {
+        discoverExtensionThroughJsonDescriber(artifactPlugin, extensionManager);
       }
+    }
+  }
 
-      if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Discovered extension " + artifactPlugin.getArtifactName());
+  /**
+   * Looks for an extension using the {@link ArtifactPluginDescriptor#MULE_PLUGIN_JSON} file, where if available it will parse it calling the
+   * {@link XmlBasedDescriber} describer.
+   *
+   * TODO(fernandezlautaro): MULE-10876 all this code will be merged within the {@link ExtensionManagerAdapterFactory} where the {@link ExtensionManifest} will be dropped and the correct {@link Describer} will be picked up (either Java or XML based)
+   *
+   * @param artifactPlugin to introspect for the {@link ArtifactPluginDescriptor#MULE_PLUGIN_JSON} and further resources from its {@link ClassLoader}
+   * @param extensionManager object to store the generated {@link ExtensionModel} if exists
+   */
+  private void discoverExtensionThroughJsonDescriber(ArtifactPlugin artifactPlugin, ExtensionManagerAdapter extensionManager) {
+    artifactPlugin.getDescriptor().getExtensionModelDescriptorProperty().ifPresent(descriptorProperty -> {
+      if (!descriptorProperty.getId().equals(XmlBasedDescriber.DESCRIBER_ID)) {
+        // TODO(fernandezlautaro): MULE-10876 work this ID with SPI, JAVA vs XML implementation should be discovered and validated against them
+        throw new IllegalArgumentException(format("The id '%s' does not match with the describers available to generate an ExtensionModel (working with the plugin '%s')",
+                                                  descriptorProperty.getId(), artifactPlugin.getDescriptor().getName()));
       }
-      ExtensionManifest extensionManifest = extensionManager.parseExtensionManifestXml(manifestUrl);
-      extensionManager.registerExtension(extensionManifest, artifactPlugin.getArtifactClassLoader().getClassLoader());
-    }
+      // TODO(fernandezlautaro): MULE-10876 the code below where it casts to a String must be moved to each concrete Describer implementation (it should receive the Map<String,Object> held by the muleArtifactDescriber.getExtensionModelDescriptor().getAttributes() messages
+      String modulePath = (String) descriptorProperty.getAttributes().get(RESOURCE_XML);
+
+      ClassLoader pluginClassLoader = artifactPlugin.getArtifactClassLoader().getClassLoader();
+      DescribingContext context = new DefaultDescribingContext(pluginClassLoader);
+      ExtensionFactory defaultExtensionFactory =
+          new DefaultExtensionFactory(new SpiServiceRegistry(), artifactPlugin.getArtifactClassLoader().getClassLoader());
+      XmlBasedDescriber describer = new XmlBasedDescriber(modulePath);
+      ExtensionModel extensionModel =
+          withContextClassLoader(pluginClassLoader,
+                                 () -> defaultExtensionFactory.createFrom(describer.describe(context), context));
+
+      extensionManager.registerExtension(extensionModel);
```

Comment:
I'm fine with this hack as long as it has a todo and gets fixed in the short term